# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/EditorBase.cpp
# Commit: edac234f1eba
# Full Hash: edac234f1eba144f9da6a10097da779749a9e90f
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-21 03:56:08
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 1: Make `TextEditor` and `HTMLEditor` implement `EditorBase::InitEditorContentAndSelection` by themselves r=m_kato
#   
#   The method is enough simple, and uses bad cast from point of view of OOP.
#   Therefore, this patch make the sub classes implement the method only for each.
#   
# ==============================================================================

diff -r 4ec21c88e8c9 -r edac234f1eba editor/libeditor/EditorBase.cpp
--- a/editor/libeditor/EditorBase.cpp	Tue Sep 20 23:29:09 2022 +0000
+++ b/editor/libeditor/EditorBase.cpp	Wed Sep 21 00:20:26 2022 +0000
@@ -376,46 +376,6 @@
   return NS_OK;
 }
 
-nsresult EditorBase::InitEditorContentAndSelection() {
-  MOZ_ASSERT(IsEditActionDataAvailable());
-
-  if (IsTextEditor()) {
-    MOZ_TRY(EnsureEmptyTextFirstChild());
-  } else {
-    nsresult rv = MOZ_KnownLive(AsHTMLEditor())
-                      ->MaybeCreatePaddingBRElementForEmptyEditor();
-    if (NS_FAILED(rv)) {
-      NS_WARNING(
-          "HTMLEditor::MaybeCreatePaddingBRElementForEmptyEditor() failed");
-      return rv;
-    }
-  }
-
-  // If the selection hasn't been set up yet, set it up collapsed to the end of
-  // our editable content.
-  // XXX I think that this shouldn't do it in `HTMLEditor` because it maybe
-  //     removed by the web app and if they call `Selection::AddRange()`,
-  //     it may cause multiple selection ranges.
-  if (!SelectionRef().RangeCount()) {
-    nsresult rv = CollapseSelectionToEndOfLastLeafNode();
-    if (MOZ_UNLIKELY(NS_FAILED(rv))) {
-      NS_WARNING("EditorBase::CollapseSelectionToEndOfLastLeafNode() failed");
-      return rv;
-    }
-  }
-
-  if (IsInPlaintextMode() && !IsSingleLineEditor()) {
-    nsresult rv = EnsurePaddingBRElementInMultilineEditor();
-    if (NS_FAILED(rv)) {
-      NS_WARNING(
-          "EditorBase::EnsurePaddingBRElementInMultilineEditor() failed");
-      return rv;
-    }
-  }
-
-  return NS_OK;
-}
-
 nsresult EditorBase::PostCreateInternal() {
   MOZ_ASSERT(IsEditActionDataAvailable());
 